adapters/repos/db/index.go

```diff
@@ -547,6 +551,23 @@ func (i *Index) updateInvertedIndexConfig(ctx context.Context,
 	return nil
 }
 
+func (i *Index) updateAsyncReplication(ctx context.Context, enabled bool) error {
+	// TODO: this method as many others are not thread-safe
```

Comment:
Can we document this in an issue somewhere, if not already done?

Reply:
fixed